src/x86/ffi.c

```diff
@@ -176,7 +176,11 @@ ffi_prep_cif_machdep(ffi_cif *cif)
       bytes = FFI_ALIGN (bytes, t->alignment);
       bytes += FFI_ALIGN (t->size, FFI_SIZEOF_ARG);
     }
+#if defined(_WIN32) && defined(__clang__)
```

Comment:
This is a weird check.  The ABI that clang uses should match what MSVC does.  If that is not the case, that is a clang bug.

Reply:
My reading of the ABI spec is that Win32 32 bit stack is 32 bit aligned. I don't think this bug is specific to clang i.e it's not a clang bug.

However: I am not in a position to verify this fix on anything other than clang-cl on Windows, so I made the guard specific to the environment that I had operationally verified.

Reply:
I doubt there is any stack alignment on NT/x86. (Usually 16 on NT/amd64 but this is of course repeatedly briefly violated, and can be violated indefinitely in leaf functions or known-call-graphs. Presumably it remains at least 8 but even that could probably be violated successfully.)